### PR TITLE
[student-libraries] Allow renaming a library when creating/publishing it

### PR DESCRIPTION
**Pull request by jmkulwik**
# Description
OLD:
image

NEW:
image

## Links

- [jira](https://codedotorg.atlassian.net/browse/STAR-927)
- [spec](https://docs.google.com/document/d/1VxMmdHdIeBWCXKofkjEy9MuWcrzeEW3Zt-Ir8pINYdA/edit)
- [all relevant PRs](https://github.com/pulls?utf8=%E2%9C%93&q=is%3Apr+author%3Ajmkulwik+archived%3Afalse+student+libraries)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/i18n/common/en_us.json [hunk 1]
```diff
@@ -819,11 +819,14 @@
   "levelStatus": "Level Status",
   "levelType": "Level Type",
   "libraryCodeError": "We can't publish your library because there is an error in the code. Go look for the square red error indicator and fix the errors.",
+  "libraryDescriptionPlaceholder": "Write a description of your library",
   "libraryExportNoCommentError": "This function cannot be exported until you add a comment to it.",
   "libraryExportTitle": "Export Functions as a Library",
-  "libraryName": "Library Name:",
+  "libraryName": "Library Name",
+  "libraryNameRequirements": "Your library's name must start with a capital letter and use only letters, numbers, and underscores.",
```
Comment (Hamms) [inline apps/i18n/common/en_us.json:826]: note that many languages have no concept of capital letters

File: apps/src/code-studio/components/libraries/libraryParser.js [hunk 2]
```diff
@@ -145,27 +145,38 @@ export function createLibraryJson(
...
- * with a capitalized letter.
+ * Creates a suggested name for a library by removing all illegal characters
+ * and ensuring is starts with a capitol letter.
+ * @param {string} libraryName The name of the library
+ * @returns {string} The name of the library without any illegal characters.
+ */
+export function suggestName(libraryName) {
+  let suggestedName = sanitizeName(libraryName);
+  if (suggestedName.length === 0 || !isNaN(suggestedName.charAt(0))) {
+    suggestedName = 'Lib' + suggestedName;
```
Comment (epeach) [inline apps/src/code-studio/components/libraries/libraryParser.js:156]: Is this okay for non-English libraries?

**Review by epeach — APPROVED** [review]
LGTM

File: apps/i18n/common/en_us.json (hunk 1 above)
Comment (jmkulwik) [reply apps/i18n/common/en_us.json:826]: I believe this came from a desire to teach good naming conventions.

File: apps/src/code-studio/components/libraries/libraryParser.js (hunk 2 above)
Comment (jmkulwik) [reply apps/src/code-studio/components/libraries/libraryParser.js:156]: Hmmm great question. I'll circle back with the curriculum & prod teams on that one.

File: apps/i18n/common/en_us.json (hunk 1 above)
Comment (jmkulwik) [reply apps/i18n/common/en_us.json:826]: Oh you mean written/spoken languages - not programming languages.

**Comment by jmkulwik** [comment]
RE: Comments regarding internationalization. I've started a thread with curriculum & prod, but I don't think this should block this PR because as these issues are not introduced by this PR and will be straightforward to update/fix when answers come in. I've scheduled that work as followup. 
Internal folks can see that item here: https://codedotorg.atlassian.net/browse/STAR-946

File: apps/i18n/common/en_us.json (hunk 1 above)
Comment (jmkulwik) [reply apps/i18n/common/en_us.json:826]: Great point. I'll circle back with curriculum and prod on that one.

**Pull request merged**